internal/controlplane/handlers_authz.go

```diff
@@ -292,32 +296,157 @@ func (s *Server) ListRoleAssignments(
 // AssignRole assigns a role to a user on a project.
 // Note that this assumes that the request has already been authorized.
 func (s *Server) AssignRole(ctx context.Context, req *minder.AssignRoleRequest) (*minder.AssignRoleResponse, error) {
-	// Request Validation
 	role := req.GetRoleAssignment().GetRole()
 	sub := req.GetRoleAssignment().GetSubject()
+	email := req.GetRoleAssignment().GetEmail()
 
-	if role == "" || sub == "" {
-		return nil, util.UserVisibleError(codes.InvalidArgument, "role and subject must be specified")
+	// Determine the target project.
+	entityCtx := engine.EntityFromContext(ctx)
+	targetProject := entityCtx.Project.ID
+
+	// Ensure user is not updating their own role
+	err := s.isUserSelfUpdating(ctx, sub, email)
+	if err != nil {
+		return nil, err
 	}
 
 	// Parse role (this also validates)
-	authzrole, err := authz.ParseRole(role)
+	authzRole, err := authz.ParseRole(role)
 	if err != nil {
 		return nil, util.UserVisibleError(codes.InvalidArgument, err.Error())
 	}
 
+	// Ensure the target project exists
+	_, err = s.store.GetProjectByID(ctx, targetProject)
+	if err != nil {
+		// If the project is not found, return an error
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, util.UserVisibleError(codes.InvalidArgument, "target project with ID %s not found", targetProject)
+		}
+		return nil, status.Errorf(codes.Internal, "error getting project: %v", err)
```

Comment:
And here I think it is a case for `InvalidArgument` as it is querying using the input from the user :D 
```suggestion
		return nil, status.Errorf(codes.InvalidArgument, "error getting project: %v", err)
```

Reply:
We have a deal on this one! 😃

Reply:
but in this case it's some unexpected error, I think Internal is OK in this case? e.g. the database connection was lost and such